collector/internal/controller.go

```diff
@@ -24,15 +24,33 @@ const (
 // Controller is a bridge between the Collector's [receiverprofiles.Profiles]
 // interface and our [internal.Controller]
 type Controller struct {
-	ctlr *controller.Controller
+	ctlr       *controller.Controller
+	onShutdown func()
+}
+
+// Option is a function that allows to configure a ControllerOption.
+type Option func(*ControllerOption)
```

Comment:
Could you configure Option as an interface, like we do in [otel-go](https://github.com/open-telemetry/opentelemetry-go/blob/6180f833bc742015cb5c98bf202971cf1328c9ec/log/logtest/recorder.go#L35-L52) for example?
That makes it more extensible.

Reply:
I'll update the code to be consistent with [otel-go](https://github.com/open-telemetry/opentelemetry-go/blob/6180f833bc742015cb5c98bf202971cf1328c9ec/log/logtest/recorder.go#L35-L52). 
For my knowledge, do you have any cases in mind where having `Option` as an interface is better than `func` for the functional option pattern?

Reply:
Fixed by https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/811/commits/7cee302f0b962fb5a4082c9086963c5958e8bed2